video/out/x11_common.c

```diff
@@ -416,12 +416,14 @@ static void xrandr_read(struct vo_x11_state *x11)
         XRRProviderResources *pr = XRRGetProviderResources(x11->display, x11->rootwin);
         for (int i = 0; i < pr->nproviders; i++) {
             XRRProviderInfo *info = XRRGetProviderInfo(x11->display, r, pr->providers[i]);
-            char *amd = strcasestr(info->name, "amd");
-            char *intel = strcasestr(info->name, "intel");
-            char *nvidia = strcasestr(info->name, "nvidia");
-            char *radeon = strcasestr(info->name, "radeon");
-            x11->has_mesa = (amd || intel || radeon) ? true : false;
-            x11->has_nvidia = nvidia ? true : false;
+            struct bstr provider_name = bstr0(info->name);
+            bstr_lower(provider_name);
+            int amd = bstr_find0(provider_name, "amd");
+            int intel = bstr_find0(provider_name, "intel");
+            int nvidia = bstr_find0(provider_name, "nvidia");
+            int radeon = bstr_find0(provider_name, "radeon");
+            x11->has_mesa = !amd || !intel || !radeon;
+            x11->has_nvidia = !nvidia;
```

Comment:
and for these there's an aptly named method:
```c
            bool amd = bstr_startswith0(provider_name, "amd");
            bool intel = bstr_startswith0(provider_name, "ntel");
            bool radeon = bstr_startswith0(provider_name, "radeon");
            x11->has_mesa = amd || intel || radeon;
            x11->has_nvidia = bstr_startswith0(provider_name, "nvidia");
```

Reply:
I don't know. Is it okay to assume it's at the beginning of the string? It is for me at least but not sure if that's a rule or not.

Reply:
I assumed that was intentional since `bstr_find0(...) == 0` is the same as this

Reply:
Ah right, that's my bad. It should have been if it's `< 0`.